orchagent/orchdaemon.cpp

```diff
@@ -329,6 +330,26 @@ void OrchDaemon::start()
          * is a good chance to flush the pipeline before next select happened.
          */
         flush();
+
+        /*
+         * Asked to check warm restart readiness.
+         * Not doing this under Select::TIMEOUT condition because of
+         * the existence of finer granularity ExecutableTimer with select
+         */
+        if (gSwitchOrch->checkRestartReady())
+        {
+            bool ret = warmRestartCheck();
+            if (ret)
+            {
+                // Orchagent is ready to perform warm restart, stop processing any new db data.
+                // Should sleep here or continue handling timers and etc.??
+                if (!gSwitchOrch->checkRestartNoFreeze())
```

Comment:
>checkRestartNoFreeze [](start = 34, length = 20)

What the use case for gSwitchOrch->checkRestartReady() == true and gSwitchOrch->checkRestartNoFreeze() == true?
I think one checker is good enough?

Reply:
checkrestartready to tell orchagent to do warmRestartCheck.

checkrestartnofreeze to tell orchagent to do freeze or not.

Reply:
I can understand the name. I am asking why need check warm start but not freeze?